File: src/main/cpp/plazma/kernel/lib/data/yaml/Yaml2JsonConverter.cpp
#include "Yaml2JsonConverter.h"
#include "plazma/kernel/lib/data/yaml/yamlparser.h"
#include "plazma/kernel/lib/data/json/jsonparser.h"
#include "plazma/kernel/lib/str/strlib.h"

namespace yaml {

    Yaml2JsonConverter::Yaml2JsonConverter() {}

    Yaml2JsonConverter::~Yaml2JsonConverter() {}

    ////

    node::Node* Yaml2JsonConverter::convertDocumentNode(node::Node* original) {
        return convertNode(original); // by default
    }

    node::Node* Yaml2JsonConverter::convertNode(node::Node* original) {
        return convertDefaultNode(original); // by default
    }


    std::string Yaml2JsonConverter::toString() {
        return "Yaml2JsonConverter";
    }

    ////

    bool Yaml2JsonConverter::isObjectType(int nodeType) {
        return nodeType == yamlparser::OBJECT_TYPE;
    }

    bool Yaml2JsonConverter::isArrayType(int nodeType) {
        return nodeType == yamlparser::ARRAY_TYPE;
    }

    bool Yaml2JsonConverter::isAttributeType(int nodeType) {
        return nodeType == yamlparser::ATTRIBUTE_TYPE;
    }

    int Yaml2JsonConverter::convertNodeType(int nodeType) {
        if (nodeType == yamlparser::OBJECT_TYPE) {
            return jsonparser::OBJECT_TYPE;
        } else if (nodeType == yamlparser::ARRAY_TYPE) {
            return jsonparser::ARRAY_TYPE;
        } else if (nodeType == yamlparser::ATTRIBUTE_TYPE) {
            return jsonparser::ATTRIBUTE_TYPE;
        }
        return 0; // by default
    }

    int Yaml2JsonConverter::convertNodeSubType(int nodeSubType) {
        return 0; // by default
    }


}

File: src/main/java/plazma/lib/data/yaml/Yaml2JsonConverter.java
/*
 * Copyright (C) 2012-2023 Oleh Hapon ohapon@users.sourceforge.net
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307, USA.
 * 
 * Oleh Hapon
 * Kyiv, UKRAINE
 * ohapon@users.sourceforge.net
 */

package plazma.lib.data.yaml;

import plazma.lib.data.json.JsonParser;
import plazma.lib.data.node.BaseNodeConverter;
import plazma.lib.data.node.Node;

public class Yaml2JsonConverter extends BaseNodeConverter {

    public Node convertDocumentNode(Node original) {
        return convertNode(original); // by default
    }

    public Node convertNode(Node original) {
        return convertDefaultNode(original); // by default
    }

    ////

    public boolean isObjectType(int nodeType) {
        return nodeType == YamlParser.OBJECT_TYPE;
    }

    public boolean isArrayType(int nodeType) {
        return nodeType == YamlParser.ARRAY_TYPE;
    }

    public boolean isAttributeType(int nodeType) {
        return nodeType == YamlParser.ATTRIBUTE_TYPE;
    }

    public int convertNodeType(int nodeType) {
        if (nodeType == YamlParser.OBJECT_TYPE) {
            return JsonParser.OBJECT_TYPE;
        } else if (nodeType == YamlParser.ARRAY_TYPE) {
            return JsonParser.ARRAY_TYPE;
        } else if (nodeType == YamlParser.ATTRIBUTE_TYPE) {
            return JsonParser.ATTRIBUTE_TYPE;
        }
        return 0; // by default
    }

    public int convertNodeSubType(int nodeSubType) {
        return 0; // by default
    }

    public String toString() {
        return "Yaml2JsonConverter";
    }

}
